### PR TITLE
Adding jsonExtractScalar function to extract field from json object

### PR DESCRIPTION
**Pull request by fx19880617**
Right now we could put a JSON blob into a string field.
This udf leverages JsonPath (https://github.com/json-path/JsonPath) DSL to read from a JSON string.

1. Adding Transform function: `jsonExtractScalar`

Function Syntax:
```
jsonExtractScalar(JSON_STRING_FIELD, JSON_PATH, OUTPUT_FORMAT)
```

Sample queries:
```
Select jsonExtractScalar(myJsonMapStr,'$.k1','STRING') from myTable  where jsonExtractScalar(myJsonMapStr,'$.k1','STRING') = 'value-k1-0'";
```

```
Select sum(jsonExtractScalar(complexMapStr,'$.k4.met','INT')) from myTable group by jsonExtractScalar(complexMapStr,'$.k1','STRING')
```

2.  Adding Transform function: `jsonExtractKey` to extract the paths for a given pattern of a json.
E.g. for a given json:
```
{ "k1": "v1", "k2": "v2", "k3": "v3" }
```
The result of `jsonExtractKey(jsonField, '$.*')` will return a list a string of matched json paths pattern. In above example, the result is  `["$['k1']", "$['k2']", "$['k3']"]`

3. Normalized transform function to resolve both `jsonExtractScalar` and `json_extract_scalar` function names to the same function.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonPathTransformFunction.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    // Check that there are exactly 2 arguments
+    if (arguments.size() != 3) {
+      throw new IllegalArgumentException(
+          "Exactly 3 arguments are required for transform function: JSON_PATH(jsonFieldName, 'json_path', 'results_type')");
+    }
+
+    TransformFunction firstArgument = arguments.get(0);
+    if (firstArgument instanceof LiteralTransformFunction || !firstArgument.getResultMetadata().isSingleValue()) {
+      throw new IllegalArgumentException(
+          "The first argument of TIME_CONVERT transform function must be a single-valued column or a transform function");
```
[the remark this answers is not in the document]
Comment (fx19880617) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonPathTransformFunction.java]: copy pasted from the other function :)

**Comment by mcvsubbu** [comment]
Can you also add documentation on how json columns are to be used? I suppose the json columns cannot be used in filters (yet) right?

**Comment by fx19880617** [comment]
Will add more documents, this could be used in selection/filtering/groupby, please refer to integration tests.

Current implementation requires the column to be a string column and the content is the json string  in order to be used. I'm still investigating how to directly ingest a record/map type into the field.

**Comment by fx19880617** [comment]
> Now that we have support for udf in filter predicates, this can be an amazing feature.
> Can we change the function names to match with presto
> 
> * https://prestodb.io/docs/current/functions/json.html
> * https://docs.aws.amazon.com/athena/latest/ug/querying-JSON.html

Rename function name to `json_extract_scalar` also added a new function to extract `json_extract_key` to extract the paths for a given json.

**Comment by kishoreg** [comment]
> Can you also add documentation on how json columns are to be used? I suppose the json columns cannot be used in filters (yet) right?

We can know after we added the support for expressions in the filter predicates

**Pull request merged**